contrib/node/src/python/pants/contrib/node/targets/node_module.py

```diff
@@ -64,3 +84,15 @@ def style_ignore_path(self):
     :rtype: string
     """
     return self.payload.style_ignore_path
+
+  @property
+  def bin_executables(self):
+    """A normalized map of bin executable names and local path to an executable
+
+    :rtype: dict
+    """
+    if isinstance(self.payload.bin_executables, string_types):
+      # In this case, the package_name is the bin name
+      return { self.package_name: self.payload.bin_executables }
```

Comment:
It might make sense to move this to the `__init__` method. That way if there's an issue with constructing the string typed `bin_executables`, it'll show up when parsing the BUILD file instead of when `bin_executables` is called.

Reply:
That's a good point, will make change

Reply:
The package_name hasn't been resolved until the end of the `__init__` when it makes the calls the `super`. There could be some redesign, but I didn't want to open a can of worms with that one here

Reply:
Sounds reasonable.